addons/account/models/account_move.py

```diff
@@ -339,7 +339,7 @@ def _onchange_currency(self):
                 line.currency_id = new_currency
 
         self.line_ids._onchange_currency()
-        self._recompute_dynamic_lines()
+        self._recompute_dynamic_lines(recompute_all_taxes=True)
```

Comment:
You don't want to recompute taxes when changing the date. Instead, you should fix `_compute_invoice_taxes_by_group` that is not re-triggered for some reason.

Reply:
Indeed, I didn't think about the date, sorry by that.
_compute_invoice_taxes_by_group Is correctly triggered, but the tax_base_amount is not correct, for the moment it didn't change, and stays with the values as it was a domestic currency.